exonum_client/protobuf_provider/provider.py

```diff
@@ -0,0 +1,113 @@
+"""Protobuf sources provider module."""
+
+from typing import Dict, List
+import os
+
+from exonum_client.protobuf_loader import ProtobufProviderInterface, ProtoFile
+from exonum_client.protobuf_provider.github import _GithubProtobufProvider
+from exonum_client.protobuf_provider.filesystem import _FilesystemProtobufProvider
+
+
+class ProtobufProvider(ProtobufProviderInterface):
+    """Protobuf Provider class.
+
+    It supports two types of receiving protobuf sources:
+
+    - via filesystem,
+    - via `github`,
+    - via external "fallback" providers.
+
+    This class is designed to obtain required Protobuf sources from different places.
+    For each service, user is able to add source from which `.proto` files will be obtained.
+
+    If for some request there will be no source set, `ProtobufProvider` will try to use
```

Comment:
```suggestion
    If for a some request there will be no source set, `ProtobufProvider` will try to use
```

Reply:
"request" here syntactically equivalent to the [mass noun](https://en.wikipedia.org/wiki/Mass_noun), so I think "a" is redundant here.

Reply:
I think it is redundant